src/components/patient/timelinePatient/index.tsx

```diff
@@ -0,0 +1,180 @@
+      labelTemplate.verticalCenter = "middle";
+      // labelTemplate.fillOpacity = 0.4;
+      labelTemplate.background.fill = new am4core.InterfaceColorSet().getFor(
+        "background"
+      );
+      // labelTemplate.background.fillOpacity = 1;
+      labelTemplate.padding(7, 7, 7, 7);
+
+      let series = chart.series.push(
+        new am4plugins_timeline.CurveColumnSeries()
+      );
+      series.columns.template.height = am4core.percent(15);
+
+      series.dataFields.openDateX = "start";
+      series.dataFields.dateX = "end";
+      series.dataFields.categoryY = "type";
+      series.baseAxis = categoryAxis;
+
+      series.columns.template.propertyFields.fill = "color"; // get color from data
+      series.columns.template.propertyFields.stroke = "color";
+      series.columns.template.strokeOpacity = 0.6;
+      series.columns.template.fillOpacity = 0.6;
+      series.columns.template.strokeWidth = 10;
+
+      let imageBullet = series.bullets.push(new am4plugins_bullets.PinBullet());
+      imageBullet.locationX = 1;
+      imageBullet.clickable = true;
+      imageBullet.propertyFields.stroke = "color";
+      imageBullet.background.propertyFields.fill = "color";
+      // imageBullet.image = new am4core.Image();
+      // imageBullet.image.propertyFields.href = "icon";
+      // imageBullet.image.scale = 0.3;
+      imageBullet.propertyFields.opacity = 0.3;
+      imageBullet.circle.radius = am4core.percent(50);
+      imageBullet.dy = -5;
+      imageBullet.events.on("hit", function(ev: any) {
+        console.log(ev.target.dataItem.dataContext);
+        //TODO: update hospitalization card
+      });
+      let textBullet = series.bullets.push(new am4charts.LabelBullet());
+      textBullet.label.propertyFields.text = "title";
+      // textBullet.disabled = true;
+      textBullet.propertyFields.disabled = "textDisabled";
+      textBullet.label.strokeOpacity = 0.3;
+      textBullet.locationX = 1;
+      textBullet.dy = -80;
+      textBullet.label.textAlign = "middle";
+
+      // chart.scrollbarX = new am4core.Scrollbar();
+      // chart.scrollbarX.align = "center";
+      // chart.scrollbarX.width = am4core.percent(75);
+      // chart.scrollbarX.opacity = 1;
+      // chart.scrollbarX.start = 0;
+      // chart.scrollbarX.end = 1;
+      // chart.scrollbarX.parent = chart.bottomAxesContainer;
+
+      let cursor = new am4plugins_timeline.CurveCursor();
+      chart.cursor = cursor;
+      cursor.xAxis = dateAxis;
+      cursor.yAxis = categoryAxis;
+      cursor.lineY.disabled = true;
+      cursor.lineX.strokeDasharray = "1,4";
+      cursor.lineX.strokeOpacity = 1;
+
+      dateAxis.renderer.tooltipLocation2 = 0;
+      categoryAxis.cursorTooltipEnabled = false;
+
+      setChart(chart);
+    }
+  }, []);
+
+  return (
+    <div className="bp3-card bp3-elevation-3 timeline">
+      <H3>
+        <Icon icon={"calendar"} /> Timeline
+      </H3>
+      <div className="timelineP"> </div>
```

Comment:
Did you forget to put something in this div?
Your chart state is never used

Reply:
I'm asking myself the same question, what's the answer ? it's best to right down rather than doing it orally

Reply:
The plugin renders the timeline in a div with a given class

`      let chart = am4core.create(
        "timelineP",
        am4plugins_timeline.CurveChart
      ) as any;
`